src/FINDOLOGIC/Helpers/ResponseHelper.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace FINDOLOGIC\Helpers;
+
+class ResponseHelper
+{
+    /**
+     * Gets a property from an object.
+     *
+     * @param mixed $obj
+     * @param string $property
+     * @param null|string $type Convert value to another type. Optional.
+     * @param bool $emptyValueIsAllowed If true, values that are 0 or 0.0 are allowed.
+     *
+     * @return mixed|null Returns the property value or null if it does not exist.
+     */
+    public static function getProperty($obj, $property, $type = null, $emptyValueIsAllowed = false)
+    {
+        if (!property_exists($obj, $property)) {
+            return null;
+        }
+
+        $propertyValue = $obj->{$property};
+        if ($type === null) {
+            return $propertyValue;
+        }
+
+        settype($propertyValue, $type);
+
+        // Check for empty after the property has been casted to submitted the type.
+        if (!$emptyValueIsAllowed && self::isEmpty($propertyValue)) {
+            return null;
```

Comment:
This seems like a bug/error scenario. Is it a good idea to silently ignore it by returning `null`?

Reply:
This case may happen if a property exists, but is `''`. For example if I have a suggestion response and of course not all suggestions do have an image URL since some of them may be text suggestions or category suggestions. In this case I want to return `null` if somebody tries to actually want to get the image of the suggestion.

Reply:
Optional fields that just exist for some kinds of suggestions could be implemented by having a smart class hierarchy with different fields for whatever is needed.

For a consumer of objects of these classes it's also handy, because they don't have to null-check on their end, and they can use the type of the object to determine how to display it.